src/uucore/src/lib/features/signals.rs

```diff
@@ -196,6 +200,36 @@ pub fn signal_name_by_value(signal_value: usize) -> Option<&'static str> {
     ALL_SIGNALS.get(signal_value).copied()
 }
 
+#[cfg(unix)]
+pub fn enable_pipe_errors() -> Result<()> {
+    // SAFETY: this function is safe as long as we do not use a custom SigHandler -- we use the default one.
+    match unsafe { signal(SIGPIPE, SigDfl) } {
+        Ok(_) => Ok(()),
+        _ => Err(Error::from(ErrorKind::Other)),
+    }
```

Comment:
I'll repeat my question here and close the other comment.

Why this error handling? Are we not interested in the actual error that occurred?

Reply:
Oh, didn't see your comment. 

The following part of the manpage answers your question, I think.

The error message that we get in return really has no meaningful value. It means that sigpipe cannot be ignored in context, and the return value may be anything. 


<details><summary>Manpage</summary>
<p>

RETURN VALUE
       If the request can be honored, signal() shall return the value of func for the most recent call to signal() for the specified signal sig.  Otherwise, SIG_ERR shall be returned, and a positive value shall be stored in errno.

ERRORS
       The signal() function **shall** fail if:

       EINVAL The sig argument is not a valid signal number or an attempt is made to catch a signal that cannot be caught or ignore a signal that cannot be ignored.

**may** fail if:

       EINVAL An attempt was made to set the action to SIG_DFL for a signal that cannot be caught or ignored (or both).


</p>
</details>

Reply:
So it's just because `tee` uses `Other` as a special case for errors? Then I think we should take `tee` out of this PR and clean that up separately, because that's just a bad API.

Reply:
Alternatively, you can put the conversion to `Other` in `tee` with an additional comment.